### PR TITLE
Add new RPC endpoints debug_setHead & debug_replayBlock

### PR DESCRIPTION
**Pull request by daniellehrner**
## PR description

`debug_setHead`: Sets the current head of the local chain by block number.
`debug_replayBlock`: Reimports a block by block number, by rolling the head of the local chain back to `blockNumber` - 1 and afterwards importing the given block.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by garyschulte — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ### Additions and Improvements
 - Updated jackson-databind library to version 2.13.4.2 addressing [CVE-2022-42003](https://nvd.nist.gov/vuln/detail/CVE-2022-42003)
+- Added new RPC endpoints `debug_setHead` & `debug_replayBlock  [4580](https://github.com/hyperledger/besu/pull/4580)
```
[the remark this answers is not in the document]
Comment (daniellehrner) [reply CHANGELOG.md]: Thanks. I've moved it.

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-version=22.10.0-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (daniellehrner) [reply gradle.properties]: I've fixed it.

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/configuration/BesuNodeFactory.java [hunk 3]
```diff
@@ -521,6 +522,7 @@ public BesuNode createExecutionEngineGenesisNode(final String name, final String
             .miningEnabled()
             .jsonRpcEnabled()
             .engineRpcEnabled(true)
+            .jsonRpcDebug()
```
[the remark this answers is not in the document]
Comment (daniellehrner) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/configuration/BesuNodeFactory.java:525]: This is for the acceptance tests only. And only when testing the engine RPC endpoints. Do you think it is a problem to enable debug for those?

[the remark this answers is not in the document]
Comment (macfarla) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/configuration/BesuNodeFactory.java:525]: I think for the ATs that's ok

**Review by garyschulte — APPROVED** [review]

**Pull request merged**